### PR TITLE
Various fixes after static analysis

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

Found by:
* Phan
* Php Inspections (EA Extended)
* PHPMD

### CODE REVIEW

File: src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function __construct(Reader $reader, PropertyMetadataFactoryInterface $de
     public function create(string $resourceClass, string $property, array $options = []) : PropertyMetadata
     {
         $parentPropertyMetadata = null;
-        if (isset($this->decorated)) {
+        if ($this->decorated) {
```
Comment (dunglas) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Can even be compared explicitly to `null`.

Comment (meyerbaptiste) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: https://github.com/api-platform/core/pull/770#discussion_r80776281 😛

**Comment by dunglas** [comment]
Thanks @meyerbaptiste

File: src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php (hunk 1 above)
Comment (teohhanhui) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Shouldn't we prefer to be explicit? :disappointed:

Comment (teohhanhui) [inline src/Metadata/Property/Factory/AnnotationPropertyMetadataFactory.php]: Yes, I'm for avoiding `isset` and `empty` as much as possible.

**Pull request merged**